dadi/lib/model/connection.js

```diff
@@ -193,15 +208,38 @@ function credentials(options) {
  * @api public
  */
 module.exports = function (options) {
+
```

Comment:
Should the check for an existing connection on line 222 be moved above the `new Connection(options);` call?  If the  connection exists, no need to instantiate a new one.